### PR TITLE
Add testing for SizeLimitHandler

### PR DESCRIPTION
**Pull request by lachlan-roberts**
Add testing for the `SizeLimitHandler` parameterized over `Jetty 9.4`,  `Jetty 12 (EE8)` and `Jetty 12 (EE10)`.
Including tests for gzipped request and responses.

### CODE REVIEW

File: runtime/testapps/src/main/resources/com/google/apphosting/runtime/jetty9/sizelimitjetty94/WEB-INF/appengine-web.xml [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+ limitations under the License.
+-->
+
+<appengine-web-app xmlns="http://appengine.google.com/ns/1.0">
+  <runtime>java21</runtime>
+  <application>sizelimithandler</application>
+  <version>1</version>
+  <threadsafe>true</threadsafe>
+  <system-properties>
+    <property name="appengine.use.EE8" value="false"/>
```
[the remark this answers is not in the document]
Comment (lachlan-roberts) [reply runtime/testapps/src/main/resources/com/google/apphosting/runtime/jetty9/sizelimitjetty94/WEB-INF/appengine-web.xml:24]: should I also remove the props in the jetty94 app, and set the runtime to java8?

[the remark this answers is not in the document]
Comment (lachlan-roberts) [reply runtime/testapps/src/main/resources/com/google/apphosting/runtime/jetty9/sizelimitjetty94/WEB-INF/appengine-web.xml:24]: ohh wait nevermind this is jetty94 app

[the remark this answers is not in the document]
Comment (lachlan-roberts) [reply runtime/testapps/src/main/resources/com/google/apphosting/runtime/jetty9/sizelimitjetty94/WEB-INF/appengine-web.xml]: done

**Comment by ludoch** [comment]
Related failure? https://github.com/GoogleCloudPlatform/appengine-java-standard/actions/runs/7392509413/job/20111087201?pr=88

**Pull request merged**